DeviceAdapters/Zaber/WdiAutofocus.h

```diff
@@ -0,0 +1,77 @@
+#pragma once
+///////////////////////////////////////////////////////////////////////////////
+// FILE:          WdiAutofocus.h
+// PROJECT:       Micro-Manager
+// SUBSYSTEM:     DeviceAdapters
+//-----------------------------------------------------------------------------
+// DESCRIPTION:   Device adapter for WDI autofocus.
+//
+// AUTHOR:        Martin Zak (contact@zaber.com)
+
+// COPYRIGHT:     Zaber Technologies, 2024
+
+// LICENSE:       This file is distributed under the BSD license.
+//                License text is included with the source distribution.
+//
+//                This file is distributed in the hope that it will be useful,
+//                but WITHOUT ANY WARRANTY; without even the implied warranty
+//                of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+//
+//                IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//                CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+//                INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES.
+
+#include "Zaber.h"
+
+extern const char* g_WdiAutofocusName;
+extern const char* g_WdiAutofocusDescription;
+
+namespace WdiAutofocus_ {
+	const double xLdaNativePerMm = 1000000.0;
+}
+
+class WdiAutofocus : public CAutoFocusBase<WdiAutofocus>, public ZaberBase
+{
+public:
+	WdiAutofocus();
+	~WdiAutofocus();
+
+	// Device API
+	// ----------
+	int Initialize();
+	int Shutdown();
+	void GetName(char* name) const;
+	bool Busy();
+
+	// AutoFocus API
+	// ----------
+	virtual bool IsContinuousFocusLocked();
```

Comment:
virtual keyword is not needed, and these should be marked override if they are overriding virtual member functions declared in ZaberBase.

override isn't necessary obviously, but if you add it the compiler which check that signatures match

Reply:
fixed, thank you.